sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -126,6 +126,35 @@ private[sql] trait SQLTestUtils
   }
 
   /**
+   * Generates the requested number of temporary paths without creating the
+   * actual files/directories and passes these to the provided function 'f'.
+   * Deletes any files/directories left at these paths after 'f' returns.
```

Comment:
Creates the requested number of temporary directories, which are then passed to `f` and will be deleted after `f` returns.